app/queries/content_items_query.rb

```diff
@@ -3,14 +3,21 @@ def self.build(options)
     ContentItemsQuery.new.results(options)
   end
 
-  def results(options = {})
+  def self.build_paged(options)
+    ContentItemsQuery.new.results(options, true)
+  end
+
+  def results(options = {}, should_page = false)
```

Comment:
English, like the force, is not very strong in me... but `should_page` does not sound familiar to me. Does it read well in English? To me it sounds better `should_paginate`, but this is basically your call.